x-pack/plugins/security_solution/public/timelines/components/flyout/header/kpis.tsx

```diff
@@ -5,61 +5,95 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useMemo } from 'react';
 
-import { EuiStat, EuiFlexItem, EuiFlexGroup } from '@elastic/eui';
+import { EuiStat, EuiFlexItem, EuiFlexGroup, EuiToolTip } from '@elastic/eui';
+import numeral from '@elastic/numeral';
+import { DEFAULT_NUMBER_FORMAT } from '../../../../../common/constants';
+import { useUiSetting$ } from '../../../../common/lib/kibana';
 import { TimelineKpiStrategyResponse } from '../../../../../common/search_strategy';
 import { getEmptyValue } from '../../../../common/components/empty_value';
 import * as i18n from './translations';
 
 export const TimelineKPIs = React.memo(
   ({ kpis, isLoading }: { kpis: TimelineKpiStrategyResponse | null; isLoading: boolean }) => {
+    const kpiFormat = '0,0.[000]a';
+    const [defaultNumberFormat] = useUiSetting$<string>(DEFAULT_NUMBER_FORMAT);
+    const formattedKpis = useMemo(() => {
+      return {
+        process: kpis === null ? getEmptyValue() : numeral(kpis.processCount).format(kpiFormat),
```

Comment:
can `kpis` be defined but the specific field `kpis.processCount` not exist? Just wondering if it's better to check for the specific field?

Reply:
No, kpis can only be null if the request is pending or in an error state. If it's not null, each kpi is defaulted to 0 https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/server/search_strategy/timeline/factory/events/kpi/index.ts#L35